crates/bevy_scene/src/scene.rs

```diff
@@ -61,6 +62,15 @@ impl Scene {
 
         // Resources archetype
         for (component_id, resource_data) in self.world.storages().resources.iter() {
+            if world
+                .components()
+                .get_info(component_id)
+                .and_then(ComponentInfo::type_id)
+                .filter(|&type_id| World::is_internal_type(type_id))
```

Comment:
I think it might be more consistent to filter this resource by its `ComponentId` rather than its `TypeId`.